common/src/stack/command/stack/commands/add/pallet/__init__.py

```diff
@@ -230,23 +274,39 @@ def umount(self, iso_name, mount_point):
 
 
 	def run(self, params, args):
-		clean, stacki_pallet_dir, updatedb, run_hooks, self.username, self.password = self.fillParams([
+		clean, stacki_pallet_dir, updatedb, run_hooks, self.checksum, self.username, self.password = self.fillParams([
 			('clean', False),
 			('dir', '/export/stack/pallets'),
 			('updatedb', True),
 			('run_hooks', True),
+			('checksum', None),
 			('username', None),
 			('password', None),
 		])
 
 		# need to provide either both or none
-		if self.username or self.password and not all((self.username, self.password)):
+		if (self.username or self.password) and not all((self.username, self.password)):
 			raise UsageError(self, 'must supply a password along with the username')
 
 		clean = self.str2bool(clean)
 		updatedb = self.str2bool(updatedb)
 		run_hooks = self.str2bool(run_hooks)
 
+		if self.checksum:
```

Comment:
`add pallet` also supports non-iso pallet formats, so this should only run against iso's and should error otherwise.